build-support/bin/cherry_pick_to_one_milestone.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+set -e
+
+# CLI Args:
+#   $1 - PR Number (E.g. "12345")
+#   $2 - Milestone
+# NOTE: This script assumes you have already fetched the commit that merged the PR to `main`, along
+# with its parent.
+
+function fail {
+  printf '%s\n' "$1" >&2
+  exit "${2-1}"
+}
+
+if [[ -z $(which gh 2> /dev/null) ]]; then
+  fail "Requires the GitHub CLI: https://cli.github.com/"
+fi
+
+if [[ -n $(git status --porcelain) ]]; then
+  fail "Git working directory must be clean."
+fi
+
+PR_NUM=$1
+TARGET_MILESTONE=$2
+
+COMMIT=$(gh pr view "$PR_NUM" --json mergeCommit --jq '.mergeCommit.oid')
+if [[ -z $COMMIT ]]; then
+  fail "Wasn't able to retrieve merge commit for $PR_NUM."
+fi
+
+TITLE=$(gh pr view "$PR_NUM" --json title --jq '.title')
+CATEGORY_LABEL=$(gh pr view "$PR_NUM" --json labels --jq '.labels.[] | select(.name|test("category:.")).name')
+REVIEWERS=$(gh pr view "$PR_NUM" --json reviews --jq '.reviews.[].author.login' | sort | uniq)
+BODY_FILE=$(mktemp "/tmp/github.cherrypick.$PR_NUM.XXXXXX")
+BRANCH_NAME="cherry-pick-$PR_NUM-to-$MILESTONE"
+
+gh pr view "$PR_NUM" --json body --jq '.body' > "$BODY_FILE"
+
+PR_CREATE_CMD=(gh pr create --base "$MILESTONE" --title "$TITLE (Cherry-pick of #$PR_NUM)" --label "$CATEGORY_LABEL" --body-file "$BODY_FILE")
+while IFS= read -r REVIEWER; do PR_CREATE_CMD+=(--reviewer "$REVIEWER"); done <<< "$REVIEWERS"
+# NB: Add the author in case someone else creates the PR (like WorkerPants)
+PR_CREATE_CMD+=(--reviewer "$(gh pr view "$PR_NUM" --json author --jq '.author.login')")
+
+git fetch https://github.com/pantsbuild/pants "$MILESTONE"
```

Comment:
I think this fetch will do an unnecessary full-history fetch on CI. Maybe it can be guarded by `if [[ $CI != true ]];` since it's not needed there?

Reply:
Why isn't it needed in CI? 👀

Reply:
Either way, we can just use fetch depth one like before, right?

Reply:
Ah, sorry, I thought CI does the fetch before runnings this script, but that's just of the merge commit.

You're right: can do it with a depth. Annoyingly, I think it is better to run without the depth when running locally, to avoid breaking someone's repo by truncating history (hence the fiddle with arrays in the old script).